pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -66,7 +66,8 @@ func (h *Helper) GetDockerAuthSearchPaths(authType string) []string {
 // SetSystemContextFilePath properly seeds the container/image SystemContext
 // with the authentication file based on the format implied by the file name
 func SetSystemContextFilePath(sc *types.SystemContext, path string) {
-	if filepath.Base(path) == dockerConfigFileKey {
+	base := filepath.Base(path)
+	if base == dockerConfigFileKey || base == DockerConfigJsonKey {
```

Comment:
@gabemontero is `.dockercfg` considered the legacy auth file path by c/image?

Reply:
yes

Reply:
Makes a bit more sense to flip the logic:

```
if base == DockerConfigKey {
    sc.LegacyFormatAuthFilePath = path
    return
}
sc.AuthFilePath = path
```

Reply:
logic flipped @adambkaplan